play/src/front/Components/MapEditor/EntityEditor/ItemUpload.svelte

```diff
@@ -0,0 +1,97 @@
+<script lang="ts">
+    import { onDestroy } from "svelte";
+    import { UploadEntityMessage, UploadEntityMessage_Direction } from "@workadventure/messages";
+    import { mapEditorEntityUploadEventStore } from "../../../Stores/MapEditorStore";
+    import iconUploadWhite from "../../images/icon-upload-white.svg";
+
+    let files: FileList | undefined = undefined;
+    let fileToUpload: File | null = null;
+    let fileInput: HTMLInputElement;
+    let dropZoneRef: HTMLDivElement;
+
+    $: {
+        if (files) {
+            fileToUpload = files.item(0);
+        }
+    }
+
+    const mapEditorEntityUploadEventStoreUnsubscriber = mapEditorEntityUploadEventStore.subscribe(
+        (uploadEntityMessage) => {
+            completeAndResetUpload(uploadEntityMessage);
+        }
+    );
+
+    function completeAndResetUpload(uploadEntityMessage: UploadEntityMessage | undefined) {
+        if (uploadEntityMessage === undefined && files !== undefined) {
+            initFileUpload();
+        }
+    }
+
+    async function processFileToUpload() {
+        const fileBuffer = await fileToUpload!.arrayBuffer();
+        const fileAsUint8Array = new Uint8Array(fileBuffer);
+        mapEditorEntityUploadEventStore.set({
+            file: fileAsUint8Array,
+            direction: UploadEntityMessage_Direction.Down,
+            name: fileToUpload!.name,
+            tags: [],
+            imagePath: fileToUpload!.name,
+            color: "",
+        });
+    }
+
+    function initFileUpload() {
+        files = undefined;
+        fileInput.value = "";
+        fileToUpload = null;
+        mapEditorEntityUploadEventStore.set(undefined);
+    }
+
+    function dropHandler(event: DragEvent) {
+        const { files: filesFromDropEvent } = event.dataTransfer ?? {};
+        if (filesFromDropEvent) {
+            if (filesFromDropEvent.length > 1) {
+                console.error("Only one file is permitted");
+            } else {
+                fileToUpload = filesFromDropEvent.item(0);
+            }
+        }
+        dropZoneRef.classList.remove("tw-border-cyan-400");
+    }
+
+    onDestroy(() => {
+        mapEditorEntityUploadEventStoreUnsubscriber();
+    });
+</script>
+
+<div
+    on:drop|preventDefault|stopPropagation={dropHandler}
+    on:dragover|preventDefault={() => dropZoneRef.classList.add("tw-border-cyan-400")}
```

Comment:
It works, but you can use this to manage class through the Svelte component: `class:tw-border-cyan-400={<your variable>}`